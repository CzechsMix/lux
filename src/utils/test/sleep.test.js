// @flow
import sleep from '../sleep';

describe('util sleep()', () => {
  const amount = 500;
  let time;

  beforeEach(() => {
    time = Date.now();
  });

  it('resolves with undefined', async () => {
    expect(await sleep(amount)).toBeUndefined();
  });

  it('sleeps for the correct amount of time', async () => {
    await sleep(amount);
    expect(Date.now() - time).toBeCloseTo(amount, 25);
  });
});
