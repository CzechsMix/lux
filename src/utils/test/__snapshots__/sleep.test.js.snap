exports[`util sleep() sleeps for the correct amount of time 1`] = `
Array [
  Array [
    [Function],
    500,
  ],
]
`;
